Core/clim-basic/text-formatting.lisp

```diff
@@ -100,14 +100,14 @@
     (thunk :bottom))
   (check-type new-margins margin-spec)
   (unless (equal new-margins old-margins)
-    (slot-makunbound stream '%page-region)
-    (call-next-method new-margins stream)))
+    (call-next-method new-margins stream)
+    (slot-unbound (class-of stream) stream '%page-region)))
 
 (defgeneric invoke-with-temporary-page (stream continuation &key margins move-cursor)
   (:method ((stream standard-page-layout) continuation &key margins (move-cursor t))
-    (letf (((stream-text-margins stream) margins))
-      (multiple-value-bind (cx cy) (stream-cursor-position stream)
-        (funcall continuation stream)
+    (multiple-value-bind (cx cy) (stream-cursor-position stream)
```

Comment:
I think we  should avoid `stream-cursor-position`, `unwind-protect` and `setf stream-cursor-position` if `move-cursor` is true. I.e.

```lisp
(flet ((do-it ()
         (letf …
           (funcall continuation …))))
  (if move-cursor
      (do-it)
      (unwind-protect
        …(do-it)…)))
```